src/gui/DatabaseWidget.cpp

```diff
@@ -1520,13 +1524,34 @@ void DatabaseWidget::onGroupChanged()
 
 void DatabaseWidget::onDatabaseModified()
 {
-    if (!m_blockAutoSave && config()->get(Config::AutoSaveAfterEveryChange).toBool()) {
+    refreshSearch();
+    int autosaveDelay = m_db->metadata()->autosaveDelay() * 60 * 1000; // min to msec for QTimer
+    bool autosaveAfterEveryChangeConfig = config()->get(Config::AutoSaveAfterEveryChange).toBool();
+    if (m_autosaveDelayTimer->isActive()) {
+        // reset delay when modified
+        m_autosaveDelayTimer->start(autosaveDelay);
+        return;
+    }
+    if (autosaveDelay > 0 && autosaveAfterEveryChangeConfig) {
+        m_autosaveDelayTimer->start(autosaveDelay);
+        return;
+    }
+    if (!m_blockAutoSave && autosaveAfterEveryChangeConfig) {
+        save();
+    } else {
+        // Only block once, then reset
+        m_blockAutoSave = false;
+    }
+}
+
+void DatabaseWidget::onAutosaveDelayTimeout()
+{
+    if (!m_blockAutoSave) {
         save();
     } else {
         // Only block once, then reset
         m_blockAutoSave = false;
     }
-    refreshSearch();
```

Comment:
Where we refresh the search now? Why did you remove it?

Reply:
Moved it to the beginning of the function (line 1527) to be able to use the return in line 1533 to avoid running code not needed. The search should be refreshed on any change anyways.

Reply:
Right, seems, I missed it.